File: api-tester/src/main/java/com/revenuecat/apitester/java/TransactionAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.models.Transaction;
import com.revenuecat.purchases.models.Price;
import com.revenuecat.purchases.Store;

import java.util.Date;

@SuppressWarnings({"unused"})
final class TransactionAPI {
    static void check(final Transaction transaction) {
        final String transactionIdentifier = transaction.getTransactionIdentifier();
        final String revenuecatId = transaction.getRevenuecatId();
        final String productIdentifier = transaction.getProductIdentifier();
        final String productId = transaction.getProductId();
        final Date purchaseDate = transaction.getPurchaseDate();
        final String storeTransactionId = transaction.getStoreTransactionId();
        final Store store = transaction.getStore();
        final String displayName = transaction.getDisplayName();
        final boolean isSandbox = transaction.isSandbox();
        final Date originalPurchaseDate = transaction.getOriginalPurchaseDate();
        final Price price = transaction.getPrice();

        @SuppressWarnings("deprecation") final Transaction deprecatedTransaction = new Transaction(
                "id",
                "id",
                "product",
                "product",
                new Date(),
                "store_id",
                Store.PLAY_STORE
        );

        final Transaction newTransaction = new Transaction(
                "id",
                "id",
                "product",
                "product",
                new Date(),
                "store_id",
                Store.PLAY_STORE,
                "Display Name",
                false,
                new Date(),
                new Price("", 0, "USD")
        );
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/TransactionAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.Store
import com.revenuecat.purchases.models.Price
import com.revenuecat.purchases.models.Transaction
import java.util.Date

@Suppress("unused", "UNUSED_VARIABLE")
private class TransactionAPI {
    fun check(transaction: Transaction) {
        with(transaction) {
            val transactionIdentifier: String = transactionIdentifier
            val revenuecatId: String = revenuecatId
            val productIdentifier: String = productIdentifier
            val productId: String = productId
            val purchaseDate: Date = purchaseDate
            val storeTransactionId: String? = storeTransactionId
            val store: Store = store
            val displayName: String? = displayName
            val isSandbox: Boolean = isSandbox
            val originalPurchaseDate: Date? = originalPurchaseDate
            val price: Price? = price
        }

        // Test deprecated constructor
        @Suppress("DEPRECATION")
        val deprecatedTransaction = Transaction(
            transactionIdentifier = "id",
            revenuecatId = "id",
            productIdentifier = "product",
            productId = "product",
            purchaseDate = Date(),
            storeTransactionId = "store_id",
            store = Store.PLAY_STORE,
        )

        // Test new constructor
        val newTransaction = Transaction(
            transactionIdentifier = "id",
            revenuecatId = "id",
            productIdentifier = "product",
            productId = "product",
            purchaseDate = Date(),
            storeTransactionId = "store_id",
            store = Store.PLAY_STORE,
            displayName = "Display Name",
            isSandbox = false,
            originalPurchaseDate = Date(),
            price = null,
        )
    }
}
